flow/aggregates.py

```diff
@@ -332,18 +351,42 @@ def __call__(self, func=None):
             )
 
 
-class _AggregatesStore(Mapping):
+class _BaseAggregateStore(Mapping):
```

Comment:
I am not sure about whether we should be creating a base class for aggregate stores because of the way we use it.

Reply:
There are a few reasons why this is helpful:
- Creates a standard _interface_ for both aggregate store classes. It's an abstract class because it doesn't implement the requirements of its parent class `Mapping`, and thus `_BaseAggregateStore` cannot be instantiated.
- If we want to have a method with an implementation that is shared by both aggregate store classes, this would allow us to share that logic. For example, I can (and should) refactor `__iter__` to be a base class method that performs `yield from self.keys()`.
- This class allows us to check `isinstance(obj, _BaseAggregateStore)` if we just want _some_ aggregate store without worrying about whether it's a default or non-default.